Source/monster.cpp

```diff
@@ -2156,8 +2165,11 @@ void AiRangedAvoidance(int i, missile_id missileType, bool checkdoors, int dam,
 	int mx = monster.position.tile.x - fx;
 	int my = monster.position.tile.y - fy;
 	Direction md = GetDirection(monster.position.tile, monster.position.last);
-	if (checkdoors && monster._msquelch < UINT8_MAX)
+	if ((monster._mAi == AI_MAGMA || monster._mAi == AI_STORM || monster._mAi == AI_BONEDEMON) && monster._msquelch < UINT8_MAX)
 		MonstCheckDoors(monster);
+	int lessmissiles = (monster._mAi == AI_ACID) ? 1 : 0;
+	int dam = (monster._mAi == AI_DIABLO) ? 40 : 4;
+	missile_id missileType = GetMissileType(monster._mAi);

```

Comment:
I think it would be better if the missile type was part of the monsterData, what do you think?

Reply:
I was thinking the same when I saw this:
https://github.com/diasurgical/devilutionX/blob/99181fd70981c81d879682a7b39616515731c9d0/Source/monster.cpp#L3703-L3744

Those are MissileSpriteData though, but quite correlated imho.
Could be part of another PR maybe :)